frame/tips/src/lib.rs

```diff
@@ -103,7 +108,7 @@ pub struct OpenTip<
 	/// scheduled.
 	closes: Option<BlockNumber>,
 	/// The members who have voted for this tip. Sorted by AccountId.
-	tips: Vec<(AccountId, Balance)>,
+	tips: BoundedVec<(AccountId, Balance), LengthBoundMaximum<Tippers>>,
```

Comment:
Note that I am concerned here about having this BoundedVec controlled by `Tippers`, which is a constant used in a different pallet. Chances to make mistakes here could be high.

Reply:
I dont see where else this is used

Reply:
not sure what your response means. I am worried that this `tips` pallet has a bounded vec whose limit is the maximum length of the `Tippers`. Tippers are defined in the council pallet, which can change in size, potentially smaller.

I wonder if someone will catch that if they reduce the council size, it may corrupt existing state in the tips pallet.

Reply:
Okay, so the tight coupling here could cause issues.  
A test could help with usability, checking that `LengthBoundMaximum<Tippers>::get()` has a specific value, but does not solve the problem.

Reply:
Well, it sounds to me the proper way to fix this is to run a migration whenever the council is reduced in size -- it actually makes sense to remove the vote on the tips whenever the council is reduced in size and hence someone is evicted from being a councilor; it doesn't make sense to still count their vote afterwards.

Reply:
A migration makes sense, but it is difficult to see that only because you change the `DesiredMembers` of the elections provider, you suddenly need a migration in the `Tips` pallet…  
We would actually need some kind of `needs_migration_on_change` attribute for storage constants :thinking:

Reply:
Huh, perhaps then elections provider should provide some lifecycle hook functions so that other pallets can listen in onto the council election event? Sounds like something that would fit this situation without undergoing a migration.